### ISSUE TITLE
Indent level (parameters: first) for async arrow functions

### ISSUE THREAD

**Issue by pbadenski** [eslint/eslint#13497]
Probably the same as #10531 which was automatically closed due to inactivity 2 years ago.

**Tell us about your environment**

ESLint Version: **7.4.0**
Node Version: **14.5.0**
yarn Version: **1.22.4**
What parser (default, Babel-ESLint, etc.) are you using? **espree & @typescript-eslint/parser**

**Configuration**

```javascript
const indentOptions = {
  "FunctionDeclaration": { "parameters": "first" },
  "FunctionExpression": { "parameters": "first" },
};
const indent = ["error", 2, indentOptions];

module.exports = {
  parserOptions: {
    ecmaVersion: 2018,
    sourceType: "module"
  },
  overrides: [
    {
      files: ["*.js"],
      parser: "espree",
      rules: {
        indent: indent
      }
    }
  ]
};
```

**What did you do?**

```javascript
const foo = async (arg1,
                   arg2) =>
{
  return arg1 + arg2;
}
```

```node_modules/.bin/eslint foo.js```

**What did you expect to happen?**

Run without errors

**What actually happened?**

```bash
/private/tmp/foo/foo.js
  2:1  error  Expected indentation of 2 spaces but found 19  indent

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

**Note:** if I remove the async then the error goes away (the "parameters": "first" is honoured).

**Comment by anikethsaha** [comment]
Thanks for the report 👍 

I think this is a bug and I can reproduce this. [demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgaW5kZW50OiBbXCJlcnJvclwiLCAyLCB7IFwiRnVuY3Rpb25EZWNsYXJhdGlvblwiOiB7XCJwYXJhbWV0ZXJzXCI6IFwiZmlyc3RcIn0sXCJGdW5jdGlvbkV4cHJlc3Npb25cIjoge1wicGFyYW1ldGVyc1wiOiBcImZpcnN0XCJ9IH1dKi9cblxuY29uc3QgZm9vID0gYXN5bmMgKGFyZzEsXG4gICAgICAgICAgICAgICAgICAgYXJnMikgPT5cbntcbiAgcmV0dXJuIGFyZzEgKyBhcmcyO1xufVxuXG5hc3luYyBmdW5jdGlvbiBmKGFyZzEsXG4gICAgICAgICAgICAgICAgIGFyZzIpe1xufVxuXG5jb25zdCBmZiA9IGFzeW5jIGZ1bmN0aW9uKGFyZzEsXG4gICAgICAgICAgICAgICAgICAgICAgICAgIGFyZzIpe30iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjExLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImluZGVudCI6Mn0sImVudiI6e319fQ==)

**Comment by yeonjuan** [comment]
I agree 👍  It seems a bug to me
